docs/resource-sharing/os-backfill-containers.md

```diff
@@ -27,34 +27,146 @@ In order to configure the container, you will need:
-       --cap-add=DAC_OVERRIDE --cap-add=SETUID --cap-add=SETGID \
-       --cap-add=SYS_ADMIN --cap-add=SYS_CHROOT --cap-add=SYS_PTRACE \
-       --cap-add=CAP_DAC_READ_SEARCH \
-       -v /cvmfs:/cvmfs:shared \
-       -e TOKEN="..." \
-       -e GLIDEIN_Site="..." \
-       -e GLIDEIN_ResourceName="..." \
-       -e GLIDEIN_Start_Extra="True" \
-       -e OSG_SQUID_LOCATION="..." \
+docker run -it --rm --user osg  \
+       --privileged             \
+       -v /path/to/token:/etc/condor/tokens-orig.d/flock.opensciencegrid.org \
+       -e GLIDEIN_Site="..."                    \
+       -e GLIDEIN_ResourceName="..."            \
+       -e GLIDEIN_Start_Extra="True"            \
+       -e OSG_SQUID_LOCATION="..."              \
+       -e CVMFSEXEC_REPOS="                     \
+            oasis.opensciencegrid.org           \
+            singularity.opensciencegrid.org"    \
        opensciencegrid/osgvo-docker-pilot:release
 ```
 
-Note the additional capabilities requested in the above `docker run` allow the container to invoke `singularity` for
-user jobs; this allows the user to utilize a container for their job that is different from the pilot.
+(Replace `/path/to/token` with the location you saved the token obtained from the OSG VO administrators.)
+Privileged mode (`--privileged`) requested in the above `docker run` allows the container
+to mount CVMFS using cvmfsexec (see below) and invoke `singularity` for user jobs.
+Singularity allows the user to use their own container for their job.
+
+
+CVMFS
+-----
+
+CVMFS is a read-only remote filesystem that many OSG jobs depend on for software and data.
+Supporting CVMFS inside your container will greatly increase the types of OSG jobs you can run.
+
+There are two methods for adding CVMFS: [enabling cvmfsexec](#adding-cvmfs-using-cvmfsexec),
+or [bind-mounting CVMFS from the host](#adding-cvmfs-via-bind-mount).
+Bind-mounting CVMFS will require CVMFS to be installed on the host first,
+but the container will need somewhat fewer privileges.
+
+
+### Adding CVMFS using cvmfsexec
+
+[cvmfsexec](https://github.com/CVMFS/cvmfsexec#readme) is a tool that can be used to mount CVMFS inside the container
+without requiring CVMFS to be installed on the host.
+
+To enable cvmfsexec, specify a space-separated list of repos in the `CVMFSEXEC_REPOS` environment variable.
+We recommend the following repos:
+-   `oasis.opensciencegrid.org`
+-   `singularity.opensciencegrid.org`
+
+cvmfsexec has the following system requirements:
+
+-   On EL7, you must have kernel version >= 3.10.0-1127 (run `uname -vr` to check), and user namespaces enabled.
+    See step 1 in the
+    [Singularity Install document](https://opensciencegrid.org/docs/worker-node/install-singularity/#enabling-unprivileged-singularity)
+    for details.
+
+-   On EL8, you must have kernel version >= 4.18 (run `uname -vr` to check).
+
+See the [cvmfsexec README](https://github.com/cvmfs/cvmfsexec#readme) details.
```

Comment:
What's in there that's not in this doc? Keep in mind that I want to avoid keeping documentation in the README wherever possible

Reply:
This is a link to the _cvmfsexec_ README, not the osgvo-docker-pilot README. There are all kinds of technical details in that one that would just clutter up this doc.

Reply:
Ah righto